ROMFS/px4fmu_common/init.d/rc.mavlink

```diff
@@ -0,0 +1,207 @@
+		set MAVLINK_DEVICE_1_ARGS "-b 57600 -r 1200"
+
+		# Start MAVLink on Wifi (ESP8266 port)
+		mavlink start -d /dev/ttyS0 -b 921600 -r 20000
+	fi
+fi
+
+#
+# MAVLink onboard / TELEM2
+#
+if param compare SYS_COMPANION 10
+then
+	frsky_telemetry start -d ${MAVLINK_DEVICE_2}
+fi
+
+if param compare SYS_COMPANION 20
+then
+	set MAVLINK_DEVICE_2		"/dev/bridge0"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 57600 -m osd -r 40000"
+
+	syslink start
+fi
+
+# 19200 Baud Rate
+if param compare SYS_COMPANION 319200
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 19200 -r 1000 -f"
+fi
+if param compare SYS_COMPANION 419200
+then	
+	iridiumsbd start -d /dev/ttyS2
+	mavlink start -d /dev/iridium -b 19200 -m iridium -r 10
+fi
+if param compare SYS_COMPANION 519200
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 19200 -m minimal -r 1000"
+fi
+
+# 38400 Baud Rate
+if param compare SYS_COMPANION 338400
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 38400 -r 1000 -f"
+fi
+if param compare SYS_COMPANION 538400
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 38400 -m minimal -r 1000"
+fi
+
+# 57600 Baud Rate
+if param compare SYS_COMPANION 57600
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 57600 -m onboard -r 5000 -x -f"
+fi
+if param compare SYS_COMPANION 157600
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 57600 -m osd -r 1000"
+fi
+if param compare SYS_COMPANION 257600
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 57600 -m magic -r 5000 -x -f"
+fi
+if param compare SYS_COMPANION 357600
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 57600 -r 1000 -f"
+fi
+if param compare SYS_COMPANION 557600
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 57600 -m minimal -r 1000"
+fi
+
+if param compare SYS_COMPANION 3115200
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 115200 -r 1000 -f"
+fi
+if param compare SYS_COMPANION 5115200
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 115200 -m minimal -r 1000"
+fi
+
+# 460800 Baud Rate
+if param compare SYS_COMPANION 460800
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 460800-m onboard -r 5000 -x -f"
+fi
+
+# 921600 Buad Rate
+if param compare SYS_COMPANION 921600
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 921600 -m onboard -r 80000 -x -f"
+fi
+if param compare SYS_COMPANION 1921600
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 921600 -r 20000"
+fi
+
+# 1500000 Baud Rate
+if param compare SYS_COMPANION 1500000
+then
+	set MAVLINK_DEVICE_2		"/dev/ttyS2"
+	set MAVLINK_DEVICE_2_ARGS 	"-b 1500000 -m onboard -r 10000 -x -f"
+fi
+
+
+#
+# Start Mavlink Device 0
+#
+mavlink start -d /dev/ttyACM0 -r 800000 -m config -x
```

Comment:
Is there any reason the `-f` flag can't be enabled here by default? Or could there be a way to set custom arguments for this easily like with other devices?

Reply:
Hi @anassinator ,
I really like your question and suggestion. Thanks!  Since @dagar reduced the memory footprint I was able

Reply:
Ah, also, in response to your question, if "-f" should be enabled by default, that's simple with today's changes.  Also, if any other flags should be set by default it is simple to add them in this framework.  Here is the documentation for the mavlink argument list for your convenience: https://dev.px4.io/en/middleware/modules_communication.html

Thanks again for your review!

Reply:
Perfect! Thanks that's all I needed!

FYI Just to make sure, I noticed that `MAVLINK_DEVICE_0` switched from `/dev/ttyACM0` to `/dev/ttyS0`. Are they indeed the same thing?